content/en/docs/guides/custom-backend-metrics.md

```diff
@@ -0,0 +1,69 @@
+---
+title: Custom Backend Metrics
+description : >-
+  A mechanism in gRPC library that allows users to inject custom metrics at the
+  gRPC server side (backends). These metrics are then reported to the gRPC
+  clients where they can be consumed by the custom load balancing algorithms to
+  make traffic distribution decisions.
+---
+
+
+### Overview
+Simple load balancing decisions can be made by taking into account local or
+global knowledge of a backend’s load, for example CPU. More sophisticated
+loadbalancing decisions are possible with application specific knowledge, e.g.
+queue depth, or by combining multiple metrics.
+
+The custom backend metrics feature exposes APIs to allow users to implement
+the metrics feedback in their LB policies.
+
+### User Cases
+The feature is mainly for advanced uses cases where a custom LB policy is used
+to route traffic more smartly to a list of backend servers to improve the
+routing performance, e.g. a weighted round robin LB policy.
+
+gRPC traditionally allows users to plug in their own load balancing policies.
+For xDS users, [custom load balancing] can be configured to select the
+special LB policy.
+
+### Metrics Reporting
+gRPC use [ORCA] service and metrics standards and supports two metrics reporting 
+mechanisms:
+
+* Per-query metrics reporting: the backend server attaches the injected custom
+metrics in the trailing metadata when the corresponding RPC finishes. This is
+typically useful for short RPCs like unary calls.
+
+
+* Out-of-band metrics reporting: the backend server periodically pushes metrics
+data, e.g. cpu and memory utilization, to the client. This is useful for all
+situations: unary calls, long RPCs in streaming calls, or no RPCs. The metrics
+emitting frequency is user-configurable, and this configuration resides in
+the custom load balancing policy.
```

Comment:
Should we briefly explain how a server can push anything out-of-band? For this to happen I think the client has to open a streaming RPC on which server pushes these updates?

Reply:
no the client does not need to open an RPC, but it does require using orca API in their LB when creating subchannel. I leave these details to the examples.